tests/integration/self_update_test.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -g expected_output
+
+function self_update_test_helper()
+{
+  local update_type="$1"
+  local kw_branch="${update_type}"
+  local base_version
+  local distro
+  local container
+  local output
+  local expected_output
+  local head_commit_hash
+
+  # Get the short commit hash of the branch
+  head_commit_hash=$(git rev-parse --short origin/"${kw_branch}")
+
+  # Get the base version from the specified branch
+  base_version=$(git show "${kw_branch}":./src/VERSION | head --lines 1)
+
+  # Format the expected output
+  expected_output=$(printf '%s\nBranch: %s\nCommit: %s' "$base_version" "$kw_branch" "$head_commit_hash")
+
+  for distro in "${DISTROS[@]}"; do
+    say "Running tests on ${distro} environment..."
```

Comment:
I think this is unnecessary as it is outputted whether the test fails or succeeds. IMHO, the distro info is only relevant if the test fails (which is displayed by the assertion).